### ISSUE TITLE
remove get_hash override from CalcJob

### ISSUE THREAD

**Issue by ltalirz** [aiidateam/aiida-core#3312]
Paraphrasing @sphuber from [here](https://github.com/aiidateam/aiida-core/pull/3111#discussion_r300740070)

> In #3108 @giovannipizzi made the `get_hash` function not callable on unstored nodes, to prevent people from getting a hash, that would change once stored. The `store` method then calls the `_get_hash` method. The overridden `CalcJobNode.get_hash` however now never gets called by the `store` method and so the exclusion of the `raw_input` folder from the hash was circumvented. I think, since we solved the problem differently now, we can simply remove the `CalcJobNode.get_hash` override

I.e. the proposal is to remove this function
https://github.com/aiidateam/aiida-core/blob/b1a61ad5ba4c226bf535adb246363e636a40cf2e/aiida/orm/nodes/process/calculation/calcjob.py#L132-L134

@giovannipizzi Does this seem correct?

**Comment by sphuber** [comment]
Yes this is correct, the overridden method can and should be removed
